projects/plugins/jetpack/modules/sharedaddy/sharing-sources.php

```diff
@@ -201,19 +213,53 @@ public function get_link( $url, $text, $title, $query = '', $id = false ) {
 		 *
 		 * @param string $text Sharing service text.
 		 * @param object $this Sharing service properties.
-		 * @param int|false $id Sharing ID.
+		 * @param string|false $id Sharing ID.
 		 * @param array $args Array of sharing service options.
 		 */
 		$text = apply_filters( 'jetpack_sharing_display_text', $text, $this, $id, $args );
 
+		/**
+		 * Filter the sharing data attributes.
+		 *
+		 * @module sharedaddy
+		 *
+		 * @since 10.10.0
+		 *
+		 * @param array $data_attributes Attributes supplied from the sharing source.
+		 *                               Note that 'data-' will be prepended to all keys.
+		 * @param Sharing_Source $this Sharing source instance.
+		 * @param string|false $id Sharing ID.
+		 * @param array $args Array of sharing service options.
+		 */
+		$data_attributes = apply_filters( 'jetpack_sharing_data_attributes', $data_attributes, $this, $id, $args );
+
+		$encoded_data_attributes = '';
+		if ( ! empty( $data_attributes ) ) {
+			$encoded_data_attributes = implode(
+				' ',
+				array_map(
+					function ( $data_key, $data_value ) {
+						return sprintf(
+							'data-%s="%s"',
+							esc_html( str_replace( array( ' ', '"' ), '', $data_key ) ),
```

Comment:
Shouldn't it be `esc_attr` here as well?

```suggestion
							esc_attr( str_replace( array( ' ', '"' ), '', $data_key ) ),
```